examples/erc20/lib.rs

```diff
@@ -70,64 +90,94 @@ mod erc20 {
             instance
         }
 
+        /// Returns the total token supply.
         #[ink(message)]
         pub fn total_supply(&self) -> Balance {
             *self.total_supply
         }
 
+        /// Returns the account balance for the specified `owner`.
         #[ink(message)]
         pub fn balance_of(&self, owner: AccountId) -> Balance {
             self.balance_of_or_zero(&owner)
         }
 
+        /// Returns the amount which `spender` is still allowed to withdraw from `owner`.
         #[ink(message)]
         pub fn allowance(&self, owner: AccountId, spender: AccountId) -> Balance {
             self.allowance_of_or_zero(&owner, &spender)
```

Comment:
I think we should simply make the implementation of `allowance_of_or_zero` inplace. We did not have this inplace in the early days because it was not possible to call ink! messages internally. Nowadays I see no reason why we shouldn't have yet another function at this point.